javascript/sdk/lib/controllers/detect.ts

```diff
@@ -27,27 +28,30 @@ export default class DetectController {
 
   /** Register (or re-register) an endpoint to your account */
   async registerEndpoint(
```

Comment:
```
  /** Register (or re-register) an endpoint to your account */
  async registerEndpoint(
    { host, serial_num, edr_id = "", tags }: RegisterEndpointParams,
    options: RequestOptions = {}
  ): Promise<string> {
    const response = await this.#client.requestWithAuth("/detect/endpoint", {
      method: "POST",
      body: JSON.stringify({ id: `${host}:${serial_num}:${edr_id}`, tags }),
      ...options,
    });

    return response.text();
  }
```

ok so I am defaulting edr_id to an empty string so that it will look like "HOST:SERIAL:" as you said

Reply:
cool, all looks good to me!

only optional style thing is if you want to add question marks in register/update enpdoint to indicate which params are optional

Reply:
so we are importing the typing from our types.ts file which you can see we are referring to it at `{ host, serial_num, edr_id = "", tags }: RegisterEndpointParams,` one of my goals is to remove the types file and directly place the typing using JS-doc but thats something for the future. It's in the works!